lib/whitehall_importer/create_revision.rb

```diff
@@ -119,7 +119,12 @@ def tags(associations)
     def find_or_create_image_revisions(images)
       images.reduce([]) do |memo, image|
         already_imported = WhitehallMigration::AssetImport.find_by(original_asset_url: image["url"])
-        revision = if already_imported
+        revision = if already_imported &&
+            (already_imported.image_revision.alt_text != image["alt_text"] ||
+             already_imported.image_revision.caption != image["caption"])
+                     WhitehallImporter::CreateImageRevision
+                      .call(document_import, image, memo.map(&:filename), already_imported.image_revision.image)
+                   elsif already_imported
```

Comment:
I wonder if we can do some of this without needing to call the CreateImageRevision class as we should be able to re-use a bunch of data and skip a bunch of steps the class takes.

For example if we have the already_imported.image_revision we should be able to do:

```
new_image_revision = Image::Revision.create!(
  image: already_imported.image_revision.image
  metadata_revision: Image::MetadataRevision.new(
    caption: image["caption"],
    alt_text: image["alt_text"],
  ),
  blob_revision: already_imported.image_revision.blob_revision,
)
```

This will mean that it re-uses the same asset files

Reply:
Good point, it means I haven't needed to touch `CreateImageRevision` at all now.